pkg/oc/cli/cmd/startbuild.go

```diff
@@ -431,27 +434,9 @@ func (o *StartBuildOptions) Run() error {
 
 	// Stream the logs from the build
 	if o.Follow {
-		opts := buildapi.BuildLogOptions{
-			Follow: true,
-			NoWait: false,
-		}
-		logClient := buildclientinternalmanual.NewBuildLogClient(o.BuildClient.RESTClient(), o.Namespace)
-		for {
-			rd, err := logClient.Logs(newBuild.Name, opts).Stream()
-			if err != nil {
-				// retry the connection to build logs when we hit the timeout.
-				if oerrors.IsTimeoutErr(err) {
-					fmt.Fprintf(o.ErrOut, "timed out getting logs, retrying\n")
-					continue
-				}
-				fmt.Fprintf(o.ErrOut, "error getting logs (%v), waiting for build to complete\n", err)
-				break
-			}
-			defer rd.Close()
-			if _, err = io.Copy(o.Out, rd); err != nil {
-				fmt.Fprintf(o.ErrOut, "error streaming logs (%v), waiting for build to complete\n", err)
-			}
-			break
+		err = o.streamBuildLogs(newBuild)
+		if err != nil {
+			fmt.Fprintf(o.ErrOut, "Failed to stream the build logs - to view the logs, run oc logs build/%s\nError: %v\n", newBuild.Name, err)
 		}
```

Comment:
The next `if` just struck me, why we check `o.Follow` in the next `if` block if it's meant only for waiting for build completion? Following should be distinct from waiting from completion, no?

Reply:
yeah good point, i think we should remove the check for o.Follow below, such that if the user only specified "--follow" and we fail to get the logs (due to timeout) the command simply exits.

that said, it's a change in behavior from the current behavior which is that even if you only specify "--follow" the command will still wait for the build to complete, even if it fails to get the logs.

But i would argue the existing behavior is a bug.  If you want the command to wait, you pass --wait.  If you pass --follow, you don't get wait semantics.

Reply:
I too agree that the existing behavior is a bug (if follow fails, the command hangs), and this will now have higher user impact because the `follow` error message includes the work-around instruction. I'll remove the `o.Follow` check.